WordPress/src/main/java/org/wordpress/android/ui/mysite/SiteIconUploadViewModel.kt

```diff
@@ -0,0 +1,90 @@
+package org.wordpress.android.ui.mysite
+
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
+import org.greenrobot.eventbus.Subscribe
+import org.greenrobot.eventbus.ThreadMode
+import org.wordpress.android.analytics.AnalyticsTracker.Stat.MY_SITE_ICON_UPLOADED
+import org.wordpress.android.analytics.AnalyticsTracker.Stat.MY_SITE_ICON_UPLOAD_UNSUCCESSFUL
+import org.wordpress.android.fluxc.model.MediaModel
+import org.wordpress.android.fluxc.model.PostModel
+import org.wordpress.android.fluxc.model.SiteModel
+import org.wordpress.android.ui.mysite.SiteIconUploadViewModel.ItemUploadedModel.MediaUploaded
+import org.wordpress.android.ui.mysite.SiteIconUploadViewModel.ItemUploadedModel.PostUploaded
+import org.wordpress.android.ui.uploads.UploadService.UploadErrorEvent
+import org.wordpress.android.ui.uploads.UploadService.UploadMediaSuccessEvent
+import org.wordpress.android.util.AppLog
+import org.wordpress.android.util.AppLog.T.MAIN
+import org.wordpress.android.util.EventBusWrapper
+import org.wordpress.android.util.analytics.AnalyticsTrackerWrapper
+import org.wordpress.android.viewmodel.Event
+import javax.inject.Inject
+
+class SiteIconUploadViewModel
```

Comment:
What was the rationale behind introducing a new ViewModel just for the site icon upload? 

I don't disagree nor agree with this approach by the way, but unless I'm missing something, it seems that there is no technical obstacle that would prevent us from implementing this in `MySiteViewModel`, so I'm just trying to understand your motivations in case I have to make similar a decision.

Reply:
I wanted to extract the upload events related logic to a separate class. However, I could probably create a view lifecycle aware component that would live inside `MySiteViewModel`. It could all live directly in the `MySiteViewModel` but there will be a lot of logic there so I'm trying to pull out as much as possible. When I'm thinking about it, I think it makes sense to use this directly in the `MySiteViewModel`

Reply:
This is looking pretty good, now!